system/src/system_network_manager.cpp

```diff
@@ -365,13 +375,16 @@ void NetworkManager::transition(State state) {
     switch (state) {
         case State::DISABLED: {
             LED_SIGNAL_START(NETWORK_OFF, BACKGROUND);
-            // FIXME:
+            // FIXME: turning-off events/diagnostics
             system_notify_event(network_status, network_status_powering_off);
             system_notify_event(network_status, network_status_off);
+            NetworkDiagnostics::instance()->status(NetworkDiagnostics::TURNED_OFF);
             break;
         }
         case State::IFACE_DOWN: {
             LED_SIGNAL_START(NETWORK_ON, BACKGROUND);
+            NetworkDiagnostics::instance()->status(NetworkDiagnostics::DISCONNECTED);
+            NetworkDiagnostics::instance()->resetConnectionAttempts();
```

Comment:
Wouldn't this reset the number of connection attempts prematurely? For example, when we're resetting the network interface after a number of unsuccessful attempts to connect to the network.

Reply:
Fixed and moved to `REQUEST_DOWN` state case instead. It may not work entirely correct for cases where Ethernet is also available, but for regular devices this should be fine.